### ISSUE TITLE
computing loss for each individual instance

### ISSUE THREAD

**Issue by eaglim** [fchollet/keras#4347]
I am trying to tweak keras to generate the loss for each individual training instance in each epoch. The _fi_loop(.) function in engine/trainining.py generates average loss and accuracy for each "batch," at

      outs = f(ins_batch)

Any hints on how to get the loss value for each individual instance?

Thanks in advance

**Comment by pechersky** [comment]
This line
https://github.com/fchollet/keras/blob/master/keras/engine/training.py#L353

calculates the mean over the calculated loss over the true and predicted labels in the batch. You could make a new function wrapper instead of `weighted_objective` that returns the un-meaned tensor.

You could also just run the fit loop with a batch_size of 1.

**Comment by eaglim** [comment]
Thank you @pechersky

**Comment by eaglim** [comment]
@pechersky regarding your suggestion about computing loss for each individual instance, if I write a new function wrapper instead of `weighted_objective` [1] that returns `score_array` instead of its average `K.mean(score_array)`, how could I use it in `_fit_loop()` [2]?

[1] https://github.com/fchollet/keras/blob/8360ef3a5af9ddf517177bdc0311ae7b032ff35f/keras/engine/training.py#L325 
[2] https://github.com/fchollet/keras/blob/8360ef3a5af9ddf517177bdc0311ae7b032ff35f/keras/engine/training.py#L757

**Comment by eaglim** [comment]
Another perhaps strange idea is to do this in `_fit_loop()` : 

```
for i in range(len(self.outputs)):
	y_true = self.targets[i]
	y_pred = self.outputs[i]
	for fn in self.loss_functions:
		score_array = fn(y_true, y_pred)
```
but I don't know if it's correct or not, also `score_array` is a Tensor which I don't know how to get its value!
